Compilation fails with GCC14

Straightforward `make` would fail (in my case x86-64, Skylake). For example:

```
pstrf.c: In function 'spstrf_':
spstrf.c:830:43: error: passing argument 1 of 'dmaxloc_' from incompatible pointer type [-Wincompatible-pointer-types]
  830 |                         itemp = mymaxloc_(&work[1], &i__4, &i__5, &c__1);
      |                                           ^~~~~~~~
      |                                           |
      |                                           real * {aka float *}
spstrf.c:259:37: note: in definition of macro 'mymaxloc_'
  259 | #define mymaxloc_(w,s,e,n) dmaxloc_(w,*(s),*(e),n)
      |                                     ^
spstrf.c:367:33: note: expected 'double *' but argument is of type 'real *' {aka 'float *'}
  367 | static integer dmaxloc_(double *w, integer s, integer e, integer *n)
      |                         ~~~~~~~~^
```

From the [GCC14 porting guide](https://gcc.gnu.org/gcc-14/porting_to.html#incompatible-pointer-types):

> **Type checking on pointer types (-Werror=incompatible-pointer-types)**
> GCC no longer allows implicitly casting all pointer types to all other pointer types. This behavior is now restricted to the void * type and its qualified variations.
> 
> To fix compilation errors resulting from that, you can add the appropriate casts, and maybe consider using void * in more places (particularly for old programs that predate the introduction of void * into the C language).

--- a/Makefile.system
+++ b/Makefile.system
@@ -356,6 +356,9 @@ OBJCONV = $(CROSS_SUFFIX)objconv
 ifeq ($(NOFORTRAN), 1)
 C_LAPACK = 1
 override FEXTRALIB = 
+ifeq ($(C_COMPILER), GCC)
+CCOMMON_OPT += -Wno-error=incompatible-pointer-types
+endif
 endif
 
 ifeq ($(C_COMPILER), GCC)

--- a/cmake/system.cmake
+++ b/cmake/system.cmake
@@ -160,6 +160,12 @@ else()
   endif ()
 endif ()
 
+if (C_LAPACK)
+  if (${CMAKE_C_COMPILER_ID} STREQUAL "GNU")
+    set(CCOMMON_OPT "${CCOMMON_OPT} -Wno-error=incompatible-pointer-types")
+  endif ()
+endif ()
+
 include("${PROJECT_SOURCE_DIR}/cmake/prebuild.cmake")
 if (DEFINED TARGET)
   if (${TARGET} STREQUAL COOPERLAKE AND NOT NO_AVX512)

